virtcontainers/kata_agent.go

```diff
@@ -818,7 +818,6 @@ func (k *kataAgent) sendReq(request interface{}) (interface{}, error) {
 	if err := k.connect(); err != nil {
 		return nil, err
 	}
-	defer k.disconnect()
```

Comment:
Yes you're right but this should not be part of this PR. I think we should go first with what I proposed so that it properly covers the case of the builtin proxy, and later we can submit another PR covering the case of a long running runtime. This should be covered more neatly, by introducing a notion of short running vs long running IMO, and allowing the current use case to still use `disconnect()`. I don't like that we implicitly handle the disconnection because the process ends.